superset/commands/database/test_connection.py

```diff
@@ -225,6 +225,10 @@ def ping(engine: Engine) -> bool:
             # bubble up the exception to return proper status code
             raise
         except Exception as ex:
+            if database.is_oauth2_enabled() and database.db_engine_spec.needs_oauth2(
```

Comment:
BigQuery raises an error when running `create_engine()` necessitating another try/except block to look for the `needs_oauth2(ex)` exception. Most DB's allow for an engine to be created without valid creds, and instead raises the exception on `engine.connect()`